src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/MtAmazonDynamoDbBySharedTable.java

```diff
@@ -158,6 +158,16 @@ protected MtAmazonDynamoDbContextProvider getMtContext() {
      */
     @Override
     public BatchGetItemResult batchGetItem(BatchGetItemRequest unqualifiedBatchGetItemRequest) {
+        // validate
+        unqualifiedBatchGetItemRequest.getRequestItems().values().forEach(keysAndAttributes -> {
+            checkArgument(keysAndAttributes.getAttributesToGet() == null,
+                "attributesToGet are not supported on BatchGetItemRequest's");
+            checkArgument(keysAndAttributes.getProjectionExpression() == null,
+                "projectionExpression is not supported on BatchGetItemRequest's");
+            checkArgument(keysAndAttributes.getExpressionAttributeNames() == null,
+                "expressionAttributeNames are not supported on BatchGetItemRequest's");
+        });
+
```

Comment:
This is an unrelated change.  The `KeysAndAttributes` object that gets return as part of the unprocessedKeys has 5 things ... keys, consistentRead flag, attributesToGet, projectionExpression, and expressionAttributeNames.  Batch get tests aren't testing the latter 4.  I suspect regular get tests aren't either.  I think it's better to fail than to silently ignore them.  Maybe I should add the same validation to the non-batch get?  Should I disallow consistentRead?  I chose to support it because it's a flag that I can copy over.  But there's no test for it yet.

Reply:
OK, I am validating that there is no consistentRead set in addition to attributesToGet, projectionExpression, and expressionAttributeNames.  And I'm validating the same in `getItem()`.